packages/library-base/src/commonMain/kotlin/io/realm/kotlin/log/RealmLog.kt

```diff
@@ -0,0 +1,365 @@
+    /**
+     * Add a logger that will be notified on log events that are equal to or exceed the currently
+     * configured [level].
+     *
+     * @param logger logger to add.
+     */
+    public fun add(logger: RealmLogger) {
+        loggersMutex.withLock {
+            loggers.add(logger)
+        }
+    }
+
+    /**
+     * Removes the given logger if possible.
+     *
+     * @param logger logger that should be removed.
+     * @return `true` if the logger was removed, `false` if it wasn't registered.
+     */
+    public fun remove(logger: RealmLogger): Boolean {
+        loggersMutex.withLock {
+            val updatedLoggers = MutableList(loggers.size) { loggers[it] }
+            return updatedLoggers.remove(logger).also {
+                loggers = updatedLoggers
+            }
+        }
+    }
+
+    /**
+     * Removes all loggers. The default system logger will be removed as well unless
+     * [removeDefaultSystemLogger] is set to `false`. [addDefaultSystemLogger] can be used
+     * to add the default logger again if it was removed.
+     *
+     * @param removeDefaultSystemLogger whether or not to also remove the default system logger.
+     * @return `true` will be returned if one or more loggers were removed, `false` if no loggers were
+     * removed.
+     */
+    public fun removeAll(removeDefaultSystemLogger: Boolean = true): Boolean {
+        loggersMutex.withLock {
+            val updatedLoggers = MutableList(loggers.size) { loggers[it] }
+            return updatedLoggers.removeAll {
+                if (!removeDefaultSystemLogger) {
+                    it != systemLoggerInstalled
+                } else {
+                    true
+                }
+            }.also {
+                if (removeDefaultSystemLogger) {
+                    systemLoggerInstalled = null
+                }
+                loggers = updatedLoggers
+            }
+        }
+    }
+
+    /**
+     * Adds a default system logger. Where it report log events will depend on the system:
+     * - On Android it will go to LogCat.
+     * - On JVM it will go to std out.
+     * - On MacOS it will go to NSLog.
+     * - On iOS it will go to NSLog.
+     *
+     * @return `true` if the system logger was added, `false` if it was already present.
+     */
+    public fun addDefaultSystemLogger(): Boolean {
+        loggersMutex.withLock {
+            if (systemLoggerInstalled == null) {
+                val systemLogger = createDefaultSystemLogger(Realm.DEFAULT_LOG_TAG)
+                val updatedLoggers = MutableList(loggers.size) { loggers[it] }
+                updatedLoggers.add(systemLogger)
+                systemLoggerInstalled = systemLogger
+                loggers = updatedLoggers
+                return true
+            }
+            return false
+        }
+    }
+
+    /**
+     * Logs a [LogLevel.TRACE] event.
+     *
+     * @param throwable optional exception to log.
+     */
+    public fun trace(throwable: Throwable?) {
```

Comment:
Why are all these methods public available? Is it intended for the user to be able to use the logger implementation to log things uniformly?